### ISSUE TITLE
parsing new maps doesn't work properly

### ISSUE THREAD

**Issue by xiv1337** [AtomicBool/cs2-map-parser#9]
no i'm not talking about how you need to use vmdl_c, e.g train thinks its visible half way across the map (maybe some meshes are left out idk?)

**Comment by AtomicBool** [comment]
> no i'm not talking about how you need to use vmdl_c, e.g train thinks its visible half way across the map (maybe some meshes are left out idk?)

Does it appears rapidly? I remember some meshes is missed in the map file like boxes in mirage A

**Comment by xiv1337** [comment]
yes, also the vphys parser doesnt like the other maps (crashes and doesnt generate data)

**Comment by AtomicBool** [comment]
> yes, also the vphys parser doesnt like the other maps (crashes and doesnt generate data)

So it crashes for train? I haven't play it for months but maybe it's happening to new maps? 
Or it crashes for most maps for your laptop?

**Comment by xiv1337** [comment]
it doesn't crash on vphys_c but when parsing vmdl it does and no im not on a laptop and my pc is good

**Comment by AtomicBool** [comment]
> it doesn't crash on vphys_c but when parsing vmdl it does and no im not on a laptop and my pc is good

This repo doesn't support VMDL format. There was a post about how to change source2viewer's code to export vphys from vmdl, simple copy-paste also works. 
However, I'll take a look at the issue with Train, updating the game rn

**Comment by B1txor** [comment]
So if i understand it correctly to get the .vphys file now, i have to open the **world_physics.vmdl_c** file in source2viewer, then just copy the contents of the **PHYS** tab, and that would be the correct data for the .vphys file?

Also, does anyone have a correctly parsed .tri file of dust2? I have done the above which i think is correct, but my exported .tri file seems to either be in a wrong format, and/or not contain the triangles. I'm not sure..

**Comment by Xsintashi** [comment]
Ok so basically: All community maps and Train don't work properly or at all.
Snippet how some spots are read on train: https://youtu.be/ynzlnCqBDeY

**Comment by exp1007** [comment]
> Ok so basically: All community maps and Train don't work properly or at all. Snippet how some spots are read on train: https://youtu.be/ynzlnCqBDeY

Mesh CollisionAttributeIndex = 1

**Comment by JanEricNitschke** [comment]
> > Ok so basically: All community maps and Train don't work properly or at all. Snippet how some spots are read on train: https://youtu.be/ynzlnCqBDeY
> 
> Mesh CollisionAttributeIndex = 1

What do you mean by that? If i use the parser i get this for the triangles from train and then obviously basically everything is visible from everywhere.

![Image](https://github.com/user-attachments/assets/04bc0b6a-d4d3-42b7-8db3-a282fc97a789)

**Comment by JanEricNitschke** [comment]
Found the issue. We cant just assume that we need to use `m_nCollisionAttributeIndex = 0`. You need to first check `m_collisionAttributes[{idx}].m_CollisionGroupString` to find the `idx`s where this is `Default` or `default` and then use those indices.

**Comment by JanEricNitschke** [comment]
> [@JanEricNitschke](https://github.com/JanEricNitschke) Can u pull the code for a look

What do you mean?

**Comment by qhb1023** [comment]
@JanEricNitschke  how to fix the code in ur way

**Comment by JanEricNitschke** [comment]
@qhb1023  I wrote mine in python. I will try to make a PR to this repo.

**Comment by JanEricNitschke** [comment]
@qhb1023 These should roughly be the changes: https://github.com/AtomicBool/cs2-map-parser/pull/16

But note that i did not compile or test them, so feel free to do that and report back.
